pylucene/app/adapter.py

```diff
@@ -92,13 +93,16 @@ def index_issues(self,
         )
         issues = repo.search(
             query,
-            attributes=['key', 'summary', 'description']
+            attributes=['key', 'summary', 'description',"comments"]
```

Comment:
This will actually not work for all issues.

Some issues have a "comments" attribute, while others have a "comment" attribute. Additionally, the JSON for these attributes is not entirely compatible. 

Furthermore, the issue_db_api client library can only deal with the "comments" attribute. If you also want to support the other one, you will have to call the API directly.

Reply:
Can you explain this in details?

Reply:
The Mongo database contains essentially the raw data obtained from the Jira API. Unfortunately, the Jira API is not always consistent -- even issues in the same project might have different schemas.

For instance, some issues might look like this:
```json
{
    ... many fields omitted ...
    "comments": [list of comments]
}
```

While other look like this:

```json
{
    ... many fields omitted ...
    "comment": [list of comments]
}
```

I think in actuality, the schema is even a little more different, but I don't have the data on hand. I will have to check once I'm home whether I still have the different schemas laying around.

At the very least, this will require an upstream patch in the issues-db-api. However, more generally, if Mohamed also wants different attributes (aside from just comments) supported in the search engine, we might have to 1) discuss what is feasible to support, and 2) move to the new DB even sooner, and normalise the issue schema.

Reply:
@Ajaykumarchawla I managed to find my old data about the database schema. Please see the attached files for the exact details; Every file contains a description of the schema for issues from that particular Jira instance. Also note that these schemas are not cross-compatible; some instances have custom fields with the same name but different schemas.

Anyway, regarding comments, there are two possible entries in the schema, which are every so slightly different (see below). By default, the issue API only fetches comments from issues using the latter schema, ignoring comments in issues which use the former of the two.

(Also, note that this is not isolated to just the comments; at the very least the attachment field is also affected IIRC)

```json
{
   optional["comment"]: {
        "startAt": number,
        "total": number,
        "comments": [
            {
                "updateAuthor": {
                    "avatarUrls": {
                        "16x16": string,
                        "24x24": string,
                        "32x32": string,
                        "48x48": string,
                    },
                    "displayName": string,
                    "name": string,
                    "timeZone": string,
                    "self": string,
                    "active": boolean,
                    "key": string,
                },
                "author": {
                    "key": string,
                    "active": boolean,
                    "displayName": string,
                    "avatarUrls": {
                        "32x32": string,
                        "24x24": string,
                        "16x16": string,
                        "48x48": string,
                    },
                    "timeZone": string,
                    "self": string,
                    "name": string,
                },
                "self": string,
                "updated": string,
                "created": string,
                "body": string,
                "id": string,
            }
        ],
        "maxResults": number,
    }
} 
```

```json
{
 optional["comments"]: [
        {
            "author": {
                "active": boolean,
                "key": string,
                "name": string,
                "displayName": string,
                "timeZone": string,
                "avatarUrls": {
                    "24x24": string,
                    "16x16": string,
                    "48x48": string,
                    "32x32": string,
                },
                "self": string,
            },
            "body": string,
            "updateAuthor": {
                "displayName": string,
                "self": string,
                "key": string,
                "timeZone": string,
                "name": string,
                "active": boolean,
                "avatarUrls": {
                    "16x16": string,
                    "32x32": string,
                    "24x24": string,
                    "48x48": string,
                },
            },
            "updated": string,
            "created": string,
            "self": string,
            "id": string,
        }
    ]
}
```


[schema_Apache.txt](https://github.com/mining-design-decisions/maestro-search-engine/files/14877194/schema_Apache.txt)
[schema_JFrog.txt](https://github.com/mining-design-decisions/maestro-search-engine/files/14877195/schema_JFrog.txt)
[schema_Jira.txt](https://github.com/mining-design-decisions/maestro-search-engine/files/14877196/schema_Jira.txt)
[schema_JiraEcosystem.txt](https://github.com/mining-design-decisions/maestro-search-engine/files/14877197/schema_JiraEcosystem.txt)
[schema_MongoDB.txt](https://github.com/mining-design-decisions/maestro-search-engine/files/14877198/schema_MongoDB.txt)
[schema_RedHat.txt](https://github.com/mining-design-decisions/maestro-search-engine/files/14877199/schema_RedHat.txt)
[schema_Sonatype.txt](https://github.com/mining-design-decisions/maestro-search-engine/files/14877200/schema_Sonatype.txt)
[schema_Spring.txt](https://github.com/mining-design-decisions/maestro-search-engine/files/14877201/schema_Spring.txt)